add GADM layers

solomon-negusse opened the pull request:
<!-- Please refer to our contributing documentation for any questions on submitting a pull request, or let us know here if you need any help -->

## Pull request checklist

Please check if your PR fulfills the following requirements:
- [ ] Make sure you are requesting to pull a topic/feature/bugfix branch (right side). Don't request your master!
- [ ] Make sure you are making a pull request against the develop branch (left side). Also you should start your branch off our develop.
- [ ] Check the commit's or even all commits' message styles matches our requested structure.
- [ ] Check your code additions will fail neither code linting checks nor unit test.



## Pull request type

<!-- Please do not submit updates to dependencies unless it fixes an issue. --> 

<!-- Please try to limit your pull request to one type, submit multiple pull requests if needed. --> 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: N/A


## What is the new behavior?
<!-- Please describe the behavior or changes that are being added by this PR. -->

-
-
-

## Does this introduce a breaking change?

- [ ] Yes
- [ ] No

<!-- If this introduces a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

<!-- Any other information that is important to this PR such as screenshots of how the component looks before and after the change. -->

solomon-negusse committed:
add GADM layers

solomon-negusse committed:
Merge branch 'feature/afi_analysis' into afi/gadm_layers

solomon-negusse committed:
remove gadm value for location rows

solomon-negusse committed:
aggregated gadm adm2 partitions of dissolved area

jterry64 commented on src/main/scala/org/globalforestwatch/layers/GADMadm0.scala:
```diff
@@ -3,16 +3,264 @@ package org.globalforestwatch.layers
 import org.globalforestwatch.grids.GridTile
 
 case class GADMadm0(gridTile: GridTile, kwargs: Map[String, Any]) 
```
It's not part of this PR per se, but for some reason I feel something is off about the casing... should ie be like GADMAdm0? GadmAdm0? It's hard with acronyms.

jterry64 commented on src/main/scala/org/globalforestwatch/summarystats/afi/AFiAnalysis.scala:
```diff
@@ -42,17 +41,36 @@ object AFiAnalysis extends SummaryAnalysis {
+      .getFeatureDataFrame(summaryRDD, spark)
+      .withColumn("gadm_id", when(col("location_id") =!= -1, lit("")).otherwise(col("gadm_id")))
+
+    val gadmAgg = summaryDF
```
maybe I'm reading, but I think you may have gotten it reversed - for GADM we want to group by GADM id

jterry64 reviewed (changes requested):
It'd be good to see a sample output to make sure this is the right requirements

solomon-negusse committed:
have gadm layers conform to repo's variable naming pattern

solomon-negusse committed:
aggregate results from nodes

solomon-negusse committed:
Rename GADMadm1.scala to GADMAdm1.scala

Rename GADMadm0.scala to GADMAdm0.scala

Rename GADMadm2.scala to GADMAdm2.scala

remaining gadm file name changes

solomon-negusse committed:
aggregate errors

solomon-negusse commented:
@jterry64: Addressed PR comments and points we discussed during pairing.

jterry64 commented on src/main/scala/org/globalforestwatch/summarystats/afi/AFiAnalysis.scala:
```diff
@@ -42,17 +41,48 @@ object AFiAnalysis extends SummaryAnalysis {
-    val summaryDF = AFiDF.getFeatureDataFrame(summaryRDD, spark)
+    val summaryDF = AFiDF
+      .getFeatureDataFrame(summaryRDD, spark)
+      .withColumn(
```
What case here are you dealing with by checking if null is in the GADM ID?

solomon-negusse replied:
I'm doing that to consistently set the column to null (specifically doing that to merge scenarios of <ABC.null.null> and <ABC.123.null>  I encountered in an example run).

jterry64 commented on src/main/scala/org/globalforestwatch/summarystats/afi/AFiAnalysis.scala:
```diff
@@ -42,17 +41,48 @@ object AFiAnalysis extends SummaryAnalysis {
+    val gadmAgg = summaryDF
+      .filter($"location_id" === -1)
+      .groupBy($"list_id")
+      .agg(
```
Would it be possible to move this agg to a function we can apply to both DFs? I think they're the same. That might make it easier to read.

solomon-negusse replied:
Good call, done.

jterry64 reviewed (approved):
Looks good! Just some minor comments and questions.

solomon-negusse committed:
incapsulate dataframe aggregation in a function
